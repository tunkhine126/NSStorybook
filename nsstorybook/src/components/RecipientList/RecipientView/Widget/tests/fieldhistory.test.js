/**
 *
 * Tests for FieldHistory
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import TestingComponent from 'components/shared/Testing';
import { render } from '@testing-library/react';
// import 'jest-dom/extend-expect'; // add some helpful assertions
import FieldHistory from '../fieldHistory';

describe('<FieldHistory />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <FieldHistory date="10/12/2018" value="James" user="Jake" />
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <TestingComponent>
        <FieldHistory date="10/12/2018" value="James" user="Jake" />
      </TestingComponent>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
