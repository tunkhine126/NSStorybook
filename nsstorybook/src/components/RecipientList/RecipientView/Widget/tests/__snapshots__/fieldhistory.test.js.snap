// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FieldHistory /> Should render and match the snapshot 1`] = `
<div
  class="MuiGrid-root makeStyles-container-147 MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-center"
>
  <p
    class="MuiTypography-root makeStyles-user-150 MuiTypography-body1"
  >
    10/12/2018
     â€“ 
    Jake
  </p>
  <p
    class="MuiTypography-root makeStyles-content-148 MuiTypography-body1"
  >
    James
  </p>
</div>
`;
